constrained-voucher.mkd

```diff
@@ -943,29 +946,50 @@ key is already known to the Pledge. Still, where needed the MASA MAY use
+a certificate or certificate chain that will enable linking the signing identity to a pre-stored Trust Anchor (CA) in the Pledge.
+This approach is not recommended, because including certificates in the protected "x5bag" or "x5chain" attributes will
+significantly increase the size of the voucher which impacts operations on constrained networks.
 
-If the MASA signing key is based upon a PKI (see {{I-D.richardson-anima-masa-considerations}} Section 2.3), and the Pledge
+For example, if the MASA signing key is based upon a PKI (see {{I-D.richardson-anima-masa-considerations}} Section 2.3), and the Pledge
 only pre-stores a manufacturer (root) CA identity in its Trust Store which is not the identity that signs the voucher,
 then a certificate chain needs to be included with the voucher in order for the Pledge to validate the MASA signing CA's signature
 by validating the chain up to the CA in its Trust Store.
-
-In BRSKI CMS signed vouchers {{RFC8995}}, the CMS structure has a place for such certificates.
-In the COSE-signed constrained vouchers described in this document, the x5bag attribute {{RFC9360}} is used to contain the needed certificates to form the chain.
-A Registrar MUST NOT remove the x5bag attribute from the unprotected COSE header parameters when sending the voucher back to the Pledge.
-
-Below, an example is shown of a COSE-signed voucher. This example shows the common case where the "x5bag" attribute is not used.
+In BRSKI CMS signed vouchers {{RFC8995}}, the CMS structure has a place for such a certificate chain.
+In the COSE-signed vouchers described in this document, the "x5chain" attribute {{RFC9360}} placed in the
+COSE protected header parameters is used to contain the needed certificates for the Pledge to form the chain.
+
+To signal the complete chain of the MASA's signing identity to the Registrar, the MASA MUST include the complete
+chain of signing certificates in an "x5chain" attribute in the unprotected header of the voucher.
+This allows the Registrar to optionally validate the voucher before forwarding it to the Pledge, or to validate it for
+logging purposes.
+There is no voucher size impact of including this certificate chain in an unprotected "x5chain" attribute for
+constrained networks, because the Registrar will remove this unprotected header prior to forwarding the voucher
+response to the Pledge as defined in {{brski-extensions-registrar}}.
+
+Note that cBRSKI currently does not support signing the voucher with an RPK for which there is no corresponding
+certificate (chain).
+Therefore, if the MASA wants to sign a voucher with an RPK that is not part of any PKIX hierarchy it needs to create
+a single self-signed "placeholder" root CA certificate that uses the RPK as the public key.
+This "placeholder" certificate is then included as the sole certificate in an unprotected "x5chain" attribute,
+as defined in the previous paragraph.
```

Comment:
This makes sense from a syntax point of view: this is a good way to embed an RPK.
It's self-signed, and in the unprotected header, so it has no security.
The Pledge would already have this RPK so it doesn't care.
This item is useful for the Registrar only; but if it's unprotected, maybe it has no value to the Registrar at all?  The Registrar would have to have received the RPK via another out-of-band means.  So it this worth anything?

Reply:
If it's self-signed in a single cert, then a Registrar can readily validate the public key signature over the single (root CA) cert. It can also validate that this is the same public key that signed the Voucher. So, any tampering or substitution can be detected as long as the Registrar requires either a single self-signed cert or a chain that can be matched up to a manufacturer (pre-known, out of band, validated) trust root.

Reply:
Now added a section for validation by Registrar, to clarify this. (Can be reworked later on if needed.)